### PR TITLE
Update on scroll restoration

### PR DESCRIPTION
**Pull request by m99coder**
This is a fix for #383 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by hipdev** [comment]
Please update this PR, I am having the same issue 👍

File: src/Sticky.jsx [hunk 1]
```diff
@@ -289,6 +289,13 @@ class Sticky extends Component {
         if (prevState.status !== this.state.status && this.props.onStateChange) {
             this.props.onStateChange({status: this.state.status});
         }
+
+        // check if we are up-to-date, is triggered in case of scroll restoration
+        if (this.props.top !== prevState.top) {
+          this.updateInitialDimension();
```
Comment (redonkulus) [inline src/Sticky.jsx:295]: nit: looks like 2 spaces instead of 4, same for below line

File: src/Sticky.jsx [hunk 2]
```diff
@@ -289,6 +289,13 @@ class Sticky extends Component {
         if (prevState.status !== this.state.status && this.props.onStateChange) {
             this.props.onStateChange({status: this.state.status});
         }
+
+        // check if we are up-to-date, is triggered in case of scroll restoration
+        if (this.props.top !== prevState.top) {
```
Comment (redonkulus) [inline src/Sticky.jsx:294]: any way to write a test for this?

Comment (m99coder) [reply src/Sticky.jsx:294]: I don’t feel capable to add a test as I’m unable to grasp the general idea of how these are set up. I also believe it’s more an edge case that can be argued about logically. Do you ever expect `this.props.top` to be unequal to `prevState.top` in the `componentDidUpdate` event? I guess, no. But if this ever happens you might agree, that an “update” is necessary.

Comment (redonkulus) [reply src/Sticky.jsx:294]: @roderickhsiao @hankhsiao care to chime in here?

**Comment by homerchen19** [comment]
I'm facing the same issue too. Hopefully this PR will be merged soon 🙏

File: src/Sticky.jsx (hunk 2 above)
Comment (roderickhsiao) [reply src/Sticky.jsx:294]: Code looks fine to me.

**Comment by redonkulus** [comment]
Changed published in https://github.com/yahoo/react-stickynode/releases/tag/v3.1.1

File: src/Sticky.jsx (hunk 2 above)
Comment (unfalse) [reply src/Sticky.jsx:294]: You have missed the case when `this.props.top` is the name of class. It causes infinite updates by calling `this.setState` again and again.

Comment (m99coder) [reply src/Sticky.jsx:294]: Hi @unfalse, do you mind elaborating a bit on what you mean by “is the name of class”? Please provide a small example, if possible.

Comment (DmitrySannikov) [reply src/Sticky.jsx:294]: Hi @m99coder
The `Sticky` component allows us to provide a `top` property of type string (class of an element). If the top property is a string, it performs a `querySelector` and calculates the top position [here](https://github.com/yahoo/react-stickynode/blob/66d8ec8199adfca1d1eff40eadc7e631267b7655/src/Sticky.jsx#L71). Following this, it sets a number to the state. As a result, the `top` property in the state will always be a number, while `this.props.top` will continue to be a string.

If we provide the `top` property as a string, then the following comparison will always be true as it is comparing a string with a number: `if (this.props.top !== prevState.top) {`. This will result in an infinite loop: `this.updateInitialDimension() -> this.setState() -> componentDidUpdate() -> this.updateInitialDimension() ....`

I think it make sense to compare the current state with the prev state `if (this.state.top !== prevState.top) {`

Comment (m99coder) [reply src/Sticky.jsx:294]: Created a new PR with exactly your suggestion here: https://github.com/yahoo/react-stickynode/pull/939
